akka-docs-dev/rst/scala/code/docs/http/scaladsl/server/FileUploadExamplesSpec.scala

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package docs.http.scaladsl.server
+
+import java.io.File
+
+import akka.http.scaladsl.model.Multipart.FormData.BodyPart
+import akka.stream.io.SynchronousFileSink
+import akka.stream.scaladsl._
+import akka.http.scaladsl.model.Multipart
+
+import scala.concurrent.Future
+
+class FileUploadExamplesSpec extends RoutingSpec {
+
+  //#simple-upload
+  val simpleRoute =
+    path("upload") {
+      entity(as[Multipart.FormData]) { formData =>
+        val completed: Source[File, _] = formData.parts.mapAsync(1) {
+          case BodyPart(name, entity, _, _) =>
+            val file = File.createTempFile("uploaded", "bin")
+            val sink = SynchronousFileSink(file)
+            entity.dataBytes.runWith(sink).map(_ => file)
```

Comment:
But the metadata+body parts *is* what the below sample is (based on Johannes snippet from the mailing list).

The CSV stuff, is that related to file uploads or just a sample that should be in there, or you mean that the file will contain CSV that we will parse?